spyder/api/panel.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Copyright © <2013-2016> <Colin Duquesnoy and others, see pyqode/AUTHORS.rst>
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the panel API.
+Adapted from https://github.com/pyQode/pyqode.core/blob/master/pyqode/core/api/panel.py
+"""
+import logging
+
+from qtpy.QtWidgets import QWidget, QApplication
+from qtpy.QtGui import QBrush, QColor, QPen, QPainter
+
+from spyder.api.mode import Mode
+
+def _logger():
+    """ Returns module's logger """
+    return logging.getLogger(__name__)
```

Comment:
I haven't notice `_logger` isn't used in the files that I copy, I'm going to delete this functions.